### PR TITLE
samples: blinky_pwm: Add stm32f072b_disco support

### PR DESCRIPTION
**Pull request by HSIHAOCHAO**
This will enable the use of stm23f072b_disco on this sample.

### CODE REVIEW

**Comment by HSIHAOCHAO** [comment]
@FRASTM could you please help to review this PR? Thanks.

**Comment by FRASTM** [comment]
> @FRASTM could you please help to review this PR? Thanks.
sure.

- detail a bit more and format the PR description  : https://github.com/zephyrproject-rtos/zephyr/pull/55661#issue-1618168879

- Most of your commit could fill the boards/arm/stm32f072b_disco/stm32f072b_disco.dts
  and samples/basic/blinky_pwm/boards/stm32f072b_disco.overlay  could be limited to 
```
&pwmleds {
	status = "okay";
};

&pwm3 {
	status = "okay";
};
```
See other overlay files in the samples/basic/blinky_pwm/boards

- put your own `Copyright (c) 2023 xxxxxx`

**Comment by HSIHAOCHAO** [comment]
> > @FRASTM could you please help to review this PR? Thanks.
> > sure.
> 
> * detail a bit more and format the PR description  : [samples: blinky_pwm: Add stm32f072b_disco support #55661 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/55661#issue-1618168879)
> * Most of your commit could fill the boards/arm/stm32f072b_disco/stm32f072b_disco.dts
>   and samples/basic/blinky_pwm/boards/stm32f072b_disco.overlay  could be limited to
> 
> ```
> &pwmleds {
> 	status = "okay";
> };
> 
> &pwm3 {
> 	status = "okay";
> };
> ```
> 
> See other overlay files in the samples/basic/blinky_pwm/boards
> 
> * put your own `Copyright (c) 2023 xxxxxx`

Done. Thanks.

File: boards/arm/stm32f072b_disco/stm32f072b_disco.dts [hunk 1]
```diff
@@ -48,11 +48,20 @@
 		};
 	};
 
+	pwmleds: pwmleds {
+		compatible = "pwm-leds";
+
```
Comment (FRASTM) [inline boards/arm/stm32f072b_disco/stm32f072b_disco.dts:53]: better add 
`status = "disabled";`
since the overlay enables the node

**Review by FRASTM — CHANGES_REQUESTED** [review]
Could be nice to add the console output, something like : 
```
*** Booting Zephyr OS build zephyr-v3.3.0-985-gb91aa2e28c10 ***
PWM-based blinky
Calibrating for channel 1...
Done calibrating; maximum/minimum periods 1000000000/7812500 nsec

```

**Comment by HSIHAOCHAO** [comment]
> Could be nice to add the console output, something like :
> 
> ```
> *** Booting Zephyr OS build zephyr-v3.3.0-985-gb91aa2e28c10 ***
> PWM-based blinky
> Calibrating for channel 1...
> Done calibrating; maximum/minimum periods 1000000000/7812500 nsec
> ```

Add the console output from my test unit.

```
Welcome to minicom 2.7

OPTIONS: 
Compiled on Jan 10 2023, 19:31:48.
Port /dev/ttyUSB0, 16:24:43

Press CTRL-A Z for help on special keys

*** Booting Zephyr OS build zephyr-v3.3.0-921-ge26bf578c674 ***
PWM-based blinky
Calibrating for channel 1...
Done calibrating; maximum/minimum periods 1000000000/7812500 nsec
```

File: boards/arm/stm32f072b_disco/stm32f072b_disco.dts (hunk 1 above)
Comment (HSIHAOCHAO) [reply boards/arm/stm32f072b_disco/stm32f072b_disco.dts:54]: Done.

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by HSIHAOCHAO** [comment]
Is there still miss anything that block this PR to merge?

**Pull request merged**